### PR TITLE
[Bugfix] Fix issues for team assessments

### PR DESCRIPTION
**Pull request by lschlesinger**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I added multiple integration tests (Jest) related to the features (with a high test coverage)
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context

Recently, errors in the (team) assessment view were reported: Solves issue #3389
Solves sentry issue [ARTEMIS-10Y ](https://sentry.ase.in.tum.de/organizations/artemis/issues/1068). 
When solving these problems I had to make changes that are described in detail below.

### Description

Following changes were made:

1. For exercises which have submissions but are not due yet, instructors were allowed to view the submission and even assess them. The **warning on the submission page**, is now also shown on the team page and within the assessment editor. (This does not apply to programming exercises, they can't be assessed before due date)

e.g. https://artemistest.ase.in.tum.de/course-management/477/file-upload-exercises/7468/submissions
image

e.g. https://artemistest.ase.in.tum.de/course-management/477/file-upload-exercises/7468/submissions/437267/assessment
image

2. Recently, errors during the assessment of team exercises occurred. Instructors AND tutors could assess teams/submissions when navigating on the team page. Assessments could be started for exercises without a submitted solution and also for exercises before due date, independent of the role. When submissions were then saved/submitted errors occurred. The current behavior does also not **comply with the assessment rules implemented for individual assessment** -> there, tutors could not assess exercises before due date for example.

e.g. https://artemistest.ase.in.tum.de/course-management/477/programming-exercises/7643/teams/395
Buttons for starting/open/continuing the assessment were disabled according to the following logic:

For all team exercises:
- Exercises should have a submission with `submitted=true`

For team exercises with due date:
- Instructors can assess submission (except for programming exercise), but should be warned on the team page and in the assessment editor.
- Tutors can assess submissions only after due date

An example situation for different exercises in different states:
Tutor view vs. Instructor view (red: no assessment possible -> button disabled, green: assessment possible -> button enabled)

Bildschirmfoto 2021-05-25 um 08 47 06

3. The Team Participation Table was reduced, as there were misleading information and the whole component was not tested

### Steps for Testing

_Background Info_:
- Team exercises have to be created as such, choose the mode TEAM in the exercise edit page
- You need to add teams after creating an exercise -> create a new team or import a team of other exercises, also choose a tutor for the team; do this once before testing and use the same team (via import from exercise) for all team exercise you create for testing
- Navigation to the team page: search a team exercise in course management, (1) click on the teams button, (2) click on the link behind the team name

(1)
Bildschirmfoto 2021-05-25 um 10 16 44

(2)
Bildschirmfoto 2021-05-25 um 10 18 10

1. Log in to Artemis as instructor
2. Recreate the situations described in the description above
3. Check if the assessment action buttons are enabled/disabled accordingly

1. Log in to Artemis as tutor for the exercise
2. Recreate the situations described in the description above
3. Check if the assessment action buttons are enabled/disabled accordingly

-> send me a message, if something is unclear or leave comment

### Test Coverage

team-participation-table.component.ts 92%

### CODE REVIEW

**Review by ArtemTsurikov — COMMENTED** [review]
The buttons work properly. The only thing is that the message cutted off. (See screenshot) Tested it on ts1. 
Screenshot 2021-05-25 at 10 35 34

**Comment by lschlesinger** [comment]
> The buttons work properly. The only thing is that the message cutted off. (See screenshot) Tested it on ts1.
> Screenshot 2021-05-25 at 10 35 34

I will fix this ;)! Thanks for testing!

File: src/main/webapp/app/assessment/assessment-header/assessment-header.component.ts [hunk 1]
```diff
@@ -27,6 +28,7 @@ export class AssessmentHeaderComponent {
     @Input() exerciseDashboardLink: string[];
     @Input() canOverride: boolean;
 
+    @Input() exercise: Exercise | undefined;
```
Comment (Strohgelaender) [inline src/main/webapp/app/assessment/assessment-header/assessment-header.component.ts:31]:
```suggestion
    @Input() exercise?: Exercise;
```

File: src/main/webapp/app/exercises/shared/team/team-participation-table/team-participation-table.component.ts [hunk 2]
```diff
@@ -139,13 +141,34 @@ export class TeamParticipationTableComponent implements OnInit {
...
-        return !submission; // there is no submission yet to assess
+    isAssessmentButtonDisabled(exercise: Exercise, submission?: Submission): boolean {
+        // Non-Submitted exercises can not be assessed
+        if (submission === null || !submission?.submitted) {
+            return true;
+        }
+        // Exercises without due date can be assessed
+        if (exercise.dueDate === undefined) {
+            return false;
+        }
```
Comment (Strohgelaender) [inline src/main/webapp/app/exercises/shared/team/team-participation-table/team-participation-table.component.ts:161]:
```suggestion
        if (!submission || !submission.submitted) {
            return true;
        }
        // Exercises without due date can be assessed
        if (!exercise.dueDate) {
            return false;
        }
```
I'm not sure about the exact code, but I think we should handle both cases `null` and `undefined` likewise shown here.

Comment (lschlesinger) [reply src/main/webapp/app/exercises/shared/team/team-participation-table/team-participation-table.component.ts:161]: `!submission` will test for both, undefined and null, so I will commit your suggestions! Thank you ;)

**Review by willergeorg — APPROVED** [review]
Tested on TS1. Worked for me.

**Review by NicolasUllmann — APPROVED** [review]
Tested on Server 1. Everything worked as expected.

**Review by krusche — APPROVED** [review]

**Pull request merged**